### PR TITLE
Add Pr-Decoration Support On ADO (AST-70135)

### PR DESCRIPTION
**Pull request by miryamfoiferCX**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Add Pr-Decoration Support on ADO

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> integration test
>unit test
>E2E test
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/commands/util/pr.go [hunk 1]
```diff
@@ -226,6 +272,13 @@ func updateAPIURLForGitlabOnPrem(apiURL string) string {
 	return gitlabCloudURL
 }
 
+func updateAPIURLForAzureOnPrem(apiURL string) string {
```
Comment (AlvoBen) [inline internal/commands/util/pr.go:275]: I think you should change this function name to: getAPIUrl.
you don't use it just for onPrem and it's not necessarily being updated

File: internal/commands/util/pr.go [hunk 2]
```diff
@@ -283,6 +336,85 @@ func runPRDecorationGitlab(prWrapper wrappers.PRWrapper, policyWrapper wrappers.
...
+		return errors.New(errorAzureOnPremParams)
+	}
+	return nil
+}
+
+func createAzureNameSpace(namespaceFlag, projectNameFlag string) string {
+	return fmt.Sprintf("%s/%s", namespaceFlag, projectNameFlag)
+}
+
+func updateScmTokenForAzure(scmTokenFlag, codeRepositoryUserName string) string {
```
Comment (AlvoBen) [inline internal/commands/util/pr.go:411]: also here, this is not update, it's validate

File: internal/commands/util/pr.go [hunk 3]
```diff
@@ -283,6 +336,85 @@ func runPRDecorationGitlab(prWrapper wrappers.PRWrapper, policyWrapper wrappers.
...
+func validateAzureOnPremParameters(apiURL, codeRepositoryUserName string) error {
+	if apiURL == "" && codeRepositoryUserName != "" {
+		log.Println(errorAzureOnPremParams)
+		return errors.New(errorAzureOnPremParams)
+	}
+	return nil
+}
+
+func createAzureNameSpace(namespaceFlag, projectNameFlag string) string {
+	return fmt.Sprintf("%s/%s", namespaceFlag, projectNameFlag)
```
Comment (AlvoBen) [inline internal/commands/util/pr.go:408]: are those params flags or values? if values please remove the flag suffix, also it's better to trimmed those values

File: internal/commands/util/pr.go (hunk 2 above)
Comment (AlvoBen) [reply internal/commands/util/pr.go:411]: what is the purpose of this function?

File: internal/commands/util/pr_test.go [hunk 4]
```diff
@@ -66,3 +78,43 @@ func TestUpdateAPIURLForGitlabOnPrem_whenAPIURLIsNotSet_ShouldReturnCloudAPIURL(
...
+}
+
+func TestUpdateScmTokenForAzureOnPrem_whenUserNameNotSet_ShouldNotUpdateToken(t *testing.T) {
+	username := ""
+	expectedToken := token
+	updatedToken := updateScmTokenForAzure(token, username)
+	asserts.Equal(t, expectedToken, updatedToken)
+}
+
+func TestCreateAzureNameSpace_ShouldCreateNamespace(t *testing.T) {
```
Comment (AlvoBen) [inline internal/commands/util/pr_test.go:107]: what should be the namespace fot the values: "   organization    ", "   Project "?

File: internal/commands/util/pr.go (hunk 3 above)
Comment (miryamfoiferCX) [reply internal/commands/util/pr.go]: done

File: internal/commands/util/pr_test.go (hunk 4 above)
Comment (miryamfoiferCX) [reply internal/commands/util/pr_test.go:107]: @AlvoBen I tested it
there is no differences if we send project name with or without spaces at the start or at the end of the project name

**Review by AlvoBen — APPROVED** [review]

**Pull request merged**